src/main/java/net/minecraftforge/fml/ModLoader.java

```diff
@@ -212,7 +212,7 @@ private void accumulateErrors(List<ModLoadingException> errors) {
         if (containers.size() != modInfoMap.size()) {
             LOGGER.fatal(LOADING,"File {} constructed {} mods: {}, but had {} mods specified: {}",
                     modFile.getFilePath(),
-                    containers.size(), containers.stream().map(ModContainer::getModId).collect(Collectors.toList()),
+                    containers.size(), containers.stream().map(c -> c != null ? c.getModId() : "(null)").collect(Collectors.toList()),
```

Comment:
@cpw I'm assuming this is not meant to contain nulls here. I made this change here to allow a completely opaque crash to become a meaninful crash, in case a Mod annotation does not have a matching entry in the mods.toml file.